rules/php80/tests/Rector/NotIdentical/StrContainsRector/Fixture/preg_match.php.inc

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Rector\Php80\Tests\Rector\NotIdentical\StrContainsRector\Fixture;
+
+class PregMatchFunction
+{
+    public function run($needle, $haystack)
+    {
+        return preg_match('/' . $needle . '/', $haystack) !== 0;
```

Comment:
https://wiki.php.net/rfc/str_contains Does not mention that str_contains supports all the preg_match wildcards etc.. are you sure this works in this generic fashion?

Reply:
I never used it. It was mentioned in the RFC on internals, where I got inspired.

How would you narrow scope so it works?

Reply:
Either you know the $needle at analysis time and can check whether it contains preg-meta chars.. 

When $needle contents is unknown at analysis time, the rector should not refactor the code site

Reply:
Great. I can imagine unknown value.

But could you provide a code sample for first case, where we know the code?
I can think of:

```php
needle = 'abc'; // refactor
needle = '\d'; // skip
preg_match('/' . $needle . '/', $haystack) !== 0;

preg_match('/abc/', $haystack) !== 0; // refactor
preg_match('/\d/', $haystack) !== 0; // skip
```

Reply:
e.g. when needle is taken from a unknown source like

```
$needle = $_GET['searchTerm'];

if (preg_match($needle, $haystack)) { // skip
}
```

Reply:
I meant the first case

> Either you know the $needle at analysis time and can check whether it contains preg-meta chars..

Reply:
maybe a pretty simple heuristic could be: 

if ($needle === preg_quore($needle, $delim)) {
  // nothing was escaped, it seems to be save to use with `str_contains`
}

Reply:
We cannot run the code, not get the values if not static.

Reply:
Thinking of this, we might skip the `preg_match()` at all. It might be too ambiguous to solve correctly.

Reply:
See https://github.com/rectorphp/rector/pull/3076